src/drivers/intel/cavs/ipc.c

```diff
@@ -211,6 +212,17 @@ void ipc_platform_send_msg(struct ipc *ipc)
 
 	list_item_append(&msg->list, &ipc->shared_ctx->empty_list);
 
+	/* Is this event IPC? */
+	if (msg->tx_size == sizeof(struct sof_ipc_comp_event)) {
```

Comment:
We shouldn't identify things by comparing sizes. There may come message with the same size but f.e. callback is random data that you going to call and crash FW.

Reply:
If you want to add callback mechanism it should be generic, for any ipc.

Reply:
Yes that's fair point. I will replace that check so we rely on message header.